src/content/blog/astro-250.mdx

```diff
@@ -0,0 +1,167 @@
+
+In 2.0 we added the ability to prerender pages in SSR apps. Now in 2.5 you can do the opposite; in a mostly static site you can allow some routes to _not_ be prerendered.
+
+The new `'hybrid'` output option is what enables this. This new option will be marked as experimental for a short period of time as we stabilize edge cases. You can use it today by enabling the `hybridOutput` flag in the `experimental` section of your config and adding an adapter:
+
+**astro.config.mjs**
+
+```js
+import { defineConfig } from "astro/config"
+import nodejs from "@astrojs/node"
+
+export default defineConfig({
+	output: "hybrid",
+	adapter: nodejs(),
+	experimental: {
+		hybridOutput: true,
+	},
+})
+```
+
+Once you've done that your static site can not _opt out_ of prerendering by setting the `prerender` export of a page or endpoint to `false`:
+
+**src/pages/contact.astro**
+
+```astro
+---
+export const prerender = false
+
+if (Astro.request.method === "POST") {
+	// handle form submission
+}
+---
+
+<form method="POST">
+	<input type="text" name="name" />
+	<input type="email" name="email" />
+	<button type="submit">Submit</button>
+</form>
+```
+
+## Custom client directives (experimental)
+
+Custom client directives allow integration authors to define new `client:` directives to gain greater control over when components load.
+
+Integrations can add these through the `astro:config:setup` hook's `addClientDirective()` API. To enable this API, the user needs to enable `customClientDirectives` in the `experimental` section of their config.
+
+**astro.config.mjs**
+
+```js
+import { defineConfig } from "astro/config"
+import onClickDirective from "astro-click-directive"
+
+export default defineConfig({
+	integrations: [onClickDirective()],
+	experimental: {
+		customClientDirectives: true,
+	},
+})
+```
+
+**astro-click-directive**
+
+```js
+export default function onClickDirective() {
+	return {
+		hooks: {
+			"astro:config:setup": ({ addClientDirective }) => {
+				addClientDirective({
+					name: "click",
+					entrypoint: "astro-click-directive/click.js",
+				})
+			},
+		},
+	}
+}
+```
+
+Now you can use `client:click` on any of your framework components with full type support.
+
+```astro
+<Counter client:click />
+```
+
+See the [LINK TO DOCS](#) to learn how to build these types of integrations.
+
+## HTML minification
+
+You can now opt-in to minification of the HTML produced by your Astro components.
+
+Using the `compressHTML` option Astro will remove all whitespace.
+
+We wanted to enable minification without sacrificing performance. For server-rendered apps, doing minification on each render could be costly. With `compressHTML` your components are compressed by the Astro compiler, only once and during the build.
+
+You can enable this in your config:
+
+```js
+import { defineConfig } from "astro/config"
+export default defineConfig({
+	compressHTML: true,
+})
+```
+
+Compression occurs both in development mode and in the final build.
```

Comment:
```suggestion
Note: compression occurs both in development mode and in the final build.
```
Otherwise, this feels a little out of nowhere.

Another option, add an explanation of why/why this is helpful: 
> Compression occurs both in development mode and in the final build so that 
-  you can spot any errors or unintended rendering?
- you can check your output while developing?

Something like that.